### PR TITLE
Master.lossgt

### PR DESCRIPTION
**Pull request by Giounona**
stratified cross validation

### CODE REVIEW

File: src/ml.py [hunk 1]
```diff
@@ -5,6 +5,8 @@
 from torch.utils.data import DataLoader as TorchDataLoader
 
 from src.optimization.monitoring import APTOSMonitor
+from src.optimization.loss import QuadraticWeightedKappa
```
Comment (JakeForsey) [inline src/ml.py]: Can you delete all the broken Kappa stuff

File: src/ml.py [hunk 2]
```diff
@@ -62,7 +67,9 @@ def test(model: torch.nn.Module, test_loader: TorchDataLoader, device, monitor:
             output = model(data)
             preds_proba = F.log_softmax(output, dim=1)
 
-            loss = F.nll_loss(preds_proba, target)
+            # loss = F.nll_loss(preds_proba, target)
```
Comment (JakeForsey) [inline src/ml.py]: Can you delete all the broken Kappa stuff

File: src/optimization/hand_tuned.py [hunk 3]
```diff
@@ -52,15 +51,16 @@
             ("brightness_contrast", {"p": 0.3, "contrast_limit": 0.3, "brightness_limit": 0.3}),
             ("crop", {"p": 0.4, "min_max_height": (200, 256), "height": 256, "width": 256, "w2h_ratio": 1.0})
         ],
-        train_test_data_frames=["data/aptos2019-blindness-detection/train.csv"],
-        train_test_directories=["data/aptos2019-blindness-detection/train_images"],
+        train_test_data_frames=["/home/giotza01/datasets/aptos2019-blindness-detection/train.csv"],
```
Comment (JakeForsey) [inline src/optimization/hand_tuned.py]: Lets try and keep the paths in version control data/

File: src/optimization/loss.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import numpy as np
```
Comment (JakeForsey) [inline src/optimization/loss.py]: Delete this hot mess :D

File: train.py [hunk 5]
```diff
@@ -53,26 +59,49 @@ def run_experiment(
...
     results = []
-    for cv_iteration in range(1,  cross_validation_iterations + 1):
+
+    # Stratified ShuffleSplit cross-validator, Provides train/test indices to split data in train/test sets.
+    sss = StratifiedShuffleSplit(n_splits=cross_validation_iterations,
+                                 test_size=experiment.test_size(),
+                                 train_size=1-experiment.test_size(),
+                                 random_state=0)
+    #TODO: will probably need debugging when more than one datasets are added
+    labels = np.asarray([x for x in dfs[0]["diagnosis"]])
```
Comment (JakeForsey) [inline train.py]: Could do something like `labels = np.array([label for feature, label in dataset])` it would be slow because it would load all the images as well but thats okay as it would cache them ready for training.

File: train.py [hunk 6]
```diff
@@ -53,26 +59,49 @@ def run_experiment(
...
+                test_ds = Subset(dataset, test_index)
+                train_ds = Subset(dataset, train_index)
+            else:
+                test_size = experiment.test_size()
+                train_ds, test_ds = torch_random_split(
+                    dataset,
+                    [round((1 - test_size) * len(dataset)), round(test_size * len(dataset))]
+                )
+
+            print("train data size: {}". format(train_ds.__len__()))
```
Comment (JakeForsey) [inline train.py]: LOGGER.info instead of print

File: train.py (hunk 5 above)
Comment (Giounona) [reply train.py]: this operation gave me the image and image name but not the labels. Not sure how to make it work.
